weather_analyzer.py

```diff
@@ -0,0 +1,120 @@
+class WeatherDataAnalyzer:
+    @staticmethod
+    def calculate_extremes(year_weather_data, key, comparison):
+        """
+        Helper function to calculate extremes for a specific key (Weather Parameter).
+
+        Args:
+        - year_weather_data (str): List having whole year's weather data.
+        - key (str): Key representing the weather parameter ("Max TemperatureC",
+          "Min TemperatureC", or "Mean Humidity").
+        - comparison (function): Comparison function (e.g., lambda x, y: x > y) to
+          determine whether a value is an extreme.
+
+        Returns:
+        - tuple: Tuple containing the extreme value and the corresponding day.
+        """
+        if key == "Max TemperatureC":
+            extreme_value = float("-inf")
+        elif key == "Min TemperatureC":
+            extreme_value = float("inf")
```

Comment:
Why do we require to set these base values for calculating extremes? If we have a list of values, can't we simply ask for max or min from it?